File: apps/trading_dashboard/frontend/src/types/market.ts
export interface AssetSummary {
  symbol: string;
  dataset_count: number;
}

export interface DatasetSummary {
  id: string;
  path: string;
  relative_path: string;
  stage: string;
  source: string;
  assets: string[];
  timeframe: string | null;
  format: string;
  columns: string[];
  metadata_path: string | null;
}

export interface OHLCVCandle {
  time: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number | null;
}

export interface CatalogItem {
  name: string;
  category: string;
  dtype: string;
  dataset_id?: string | null;
}

export type FeatureCatalog = Record<string, CatalogItem[]>;

export interface TimeValuePoint {
  time: string;
  value: number | string | boolean | null;
}

export interface NamedSeries {
  series_id: string;
  source_type: string;
  points: TimeValuePoint[];
}

export interface SeriesResponse {
  series: NamedSeries[];
}

export interface TradeRecord {
  entry_time: string | null;
  exit_time: string | null;
  side: "long" | "short" | string;
  entry_price: number | null;
  exit_price: number | null;
  pnl: number | null;
  return: number | null;
  size: number | null;
}

File: apps/trading_dashboard/backend/app/schemas/market.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class AssetSummary(BaseModel):
    symbol: str
    dataset_count: int


class DatasetSummary(BaseModel):
    id: str
    path: str
    relative_path: str
    stage: str
    source: str
    assets: list[str]
    timeframe: str | None = None
    format: str
    columns: list[str]
    metadata_path: str | None = None


class OHLCVCandle(BaseModel):
    time: str
    open: float
    high: float
    low: float
    close: float
    volume: float | None = None


class CatalogItem(BaseModel):
    name: str
    category: str
    dtype: str
    dataset_id: str | None = None
    metadata: dict[str, Any] | None = None


class TimeValuePoint(BaseModel):
    time: str
    value: Any = None


class NamedSeries(BaseModel):
    series_id: str
    source_type: str
    points: list[TimeValuePoint]


class SeriesResponse(BaseModel):
    series: list[NamedSeries]

